domain/storage/state/storagepool.go

```diff
@@ -275,10 +339,11 @@ type loadStoragePoolsFunc func(ctx context.Context, tx *sqlair.TX) ([]domainstor
 
 func storagePoolsLoader(wantNames domainstorage.Names, wantProviders domainstorage.Providers) (loadStoragePoolsFunc, error) {
 	query := `
-SELECT (sp.uuid, sp.name, sp.type) AS (&StoragePool.*),
+SELECT (sp.uuid, sp.name, sp.type, spo.origin) AS (&StoragePool.*),
```

Comment:
There is currently no reason to expose the origin upwards to the caller. We don't have this concept in the params for the client. I would wait till there is an actual need for this value.

At the moment this work is about getting a single source of truth for storage pools so we can correctly use uuids to force correctness in storage creation.